### Import libraries and data

In [1]:
import pandas as pd

In [2]:
!ls data

bom.movie_gross.csv  title.akas.csv	   title.ratings.csv
name.basics.csv      title.basics.csv	   tmdb.movies.csv
rt.movie_info.tsv    title.crew.csv	   tn.movie_budgets.csv
rt.reviews.tsv	     title.principals.csv


In [3]:
# Title, Studio, Domestic Gross, Foreign Gross, Year
movie_gross_df = pd.read_csv('data/bom.movie_gross.csv')

# Name variable (nconst), Name of Crew/Principal, Primary Profession, Known for Titles (tconst)
people_info_df = pd.read_csv('data/name.basics.csv')

# ID, Rating, Genre, Director, Writer, Theater Date, Box Office, Runtime, Studio
rt_movie_info_df = pd.read_csv('data/rt.movie_info.tsv', delimiter='\t')

# ID, Rating, Fresh, Top Critic
rt_movie_reviews_df = pd.read_csv('data/rt.reviews.tsv', delimiter='\t', encoding='latin1')

# Title ID, Title, Region, Language
title_akas_df = pd.read_csv('data/title.akas.csv')

# Title ID (tconst), Primary Title, Original Title, Year, Runtime, Genres
title_basics_df = pd.read_csv('data/title.basics.csv')

# Title ID (tconst), Directors (nconst), Writers (nconst)
title_crew_df = pd.read_csv('data/title.crew.csv')

# Title ID (tconst), People (nconst), Category (primary profession)
title_principals_df = pd.read_csv('data/title.principals.csv')

# Titled ID (tconst), Rating, Num Votes
title_ratings_df = pd.read_csv('data/title.ratings.csv')

# Genre, Language, Original Title, Popularity, Date, Title, Average Rating, Num of Ratings
tmdb_movies_df = pd.read_csv('data/tmdb.movies.csv')

# Release Date, Title, Budget, Domestic Gross, Worldwide Gross
movie_budgets_df = pd.read_csv('data/tn.movie_budgets.csv')

### Splitting up the work

Matt is going to work on joining any of the dataframes with the tconst or nconst variables.

Grace will work on joining the Rotten Tomatoes dataframes.

Jeff will work on joining the rest.

### Matt's dataframes for reference:

people_info_df

title_akas_df

title_basics_df

title_crew_df

title_principals_df

title_ratings_df

### Get column names, shape, and number of unique values

In [7]:
print(people_info_df.columns)

print(title_akas_df.columns)

print(title_basics_df.columns)

print(title_crew_df.columns)

print(title_principals_df.columns)

print(title_ratings_df.columns)

Index(['nconst', 'primary_name', 'birth_year', 'death_year',
       'primary_profession', 'known_for_titles'],
      dtype='object')
Index(['title_id', 'ordering', 'title', 'region', 'language', 'types',
       'attributes', 'is_original_title'],
      dtype='object')
Index(['tconst', 'primary_title', 'original_title', 'start_year',
       'runtime_minutes', 'genres'],
      dtype='object')
Index(['tconst', 'directors', 'writers'], dtype='object')
Index(['tconst', 'ordering', 'nconst', 'category', 'job', 'characters'], dtype='object')
Index(['tconst', 'averagerating', 'numvotes'], dtype='object')


In [8]:
print(people_info_df.shape)

print(title_akas_df.shape)

print(title_basics_df.shape)

print(title_crew_df.shape)

print(title_principals_df.shape)

print(title_ratings_df.shape)

(606648, 6)
(331703, 8)
(146144, 6)
(146144, 3)
(1028186, 6)
(73856, 3)


In [9]:
print(people_info_df.shape)

print(len(pd.unique(title_akas_df['title_id'])))

print(len(pd.unique(title_basics_df['tconst'])))

print(len(pd.unique(title_crew_df['tconst'])))

print(len(pd.unique(title_principals_df['tconst'])))

print(len(pd.unique(title_ratings_df['tconst'])))

(606648, 6)
122302
146144
146144
143454
73856


### Joining the dataframes

...through a series of successive left joins on columns contain a 
'tconst' value, starting with the dataframe containing the highest number of 'tconst' items.

In [10]:
first_join = title_basics_df.merge(title_crew_df, how='left', left_on='tconst', right_on='tconst')
first_join

tconst                                primary_title  \
0       tt0063540                                    Sunghursh   
1       tt0066787              One Day Before the Rainy Season   
2       tt0069049                   The Other Side of the Wind   
3       tt0069204                              Sabse Bada Sukh   
4       tt0100275                     The Wandering Soap Opera   
...           ...                                          ...   
146139  tt9916538                          Kuambil Lagi Hatiku   
146140  tt9916622  Rodolpho Teóphilo - O Legado de um Pioneiro   
146141  tt9916706                              Dankyavar Danka   
146142  tt9916730                                       6 Gunn   
146143  tt9916754               Chico Albuquerque - Revelações   

                                     original_title  start_year  \
0                                         Sunghursh        2013   
1                                   Ashad Ka Ek Din        2019   
2                        The Other Side of the Wind        2018   
3                                   Sabse Bada Sukh        2018   
4                             La Telenovela Errante        2017   
...                                             ...         ...   
146139                          Kuambil Lagi Hatiku        2019   
146140  Rodolpho Teóphilo - O Legado de um Pioneiro        2015   
146141                              Dankyavar Danka        2013   
146142                                       6 Gunn        2017   
146143               Chico Albuquerque - Revelações        2013   

        runtime_minutes                genres            directors  \
0                 175.0    Action,Crime,Drama            nm0712540   
1                 114.0       Biography,Drama            nm0002411   
2                 122.0                 Drama            nm0000080   
3                   NaN          Comedy,Drama            nm0611531   
4                  80.0  Comedy,Drama,Fantasy  nm0765384,nm0749914   
...                 ...                   ...                  ...   
146139            123.0                 Drama            nm8185151   
146140              NaN           Documentary  nm9272490,nm9272491   
146141              NaN                Comedy            nm7764440   
146142            116.0                   NaN           nm10538612   
146143              NaN           Documentary  nm9272490,nm8349149   

                                        writers  
0       nm0023551,nm1194313,nm0347899,nm1391276  
1                                           NaN  
2                           nm0000080,nm0462648  
3                                     nm0347899  
4                           nm1360635,nm0749914  
...                                         ...  
146139            nm4843252,nm4900525,nm2679404  
146140                      nm9272490,nm9272491  
146141                                nm7933903  
146142                               nm10538612  
146143                      nm8349149,nm9272490  

[146144 rows x 8 columns]

In [11]:
second_join = first_join.merge(title_principals_df, how='left', left_on='tconst', right_on='tconst')
second_join

tconst                   primary_title  \
0        tt0063540                       Sunghursh   
1        tt0063540                       Sunghursh   
2        tt0063540                       Sunghursh   
3        tt0063540                       Sunghursh   
4        tt0063540                       Sunghursh   
...            ...                             ...   
1030871  tt9916754  Chico Albuquerque - Revelações   
1030872  tt9916754  Chico Albuquerque - Revelações   
1030873  tt9916754  Chico Albuquerque - Revelações   
1030874  tt9916754  Chico Albuquerque - Revelações   
1030875  tt9916754  Chico Albuquerque - Revelações   

                         original_title  start_year  runtime_minutes  \
0                             Sunghursh        2013            175.0   
1                             Sunghursh        2013            175.0   
2                             Sunghursh        2013            175.0   
3                             Sunghursh        2013            175.0   
4                             Sunghursh        2013            175.0   
...                                 ...         ...              ...   
1030871  Chico Albuquerque - Revelações        2013              NaN   
1030872  Chico Albuquerque - Revelações        2013              NaN   
1030873  Chico Albuquerque - Revelações        2013              NaN   
1030874  Chico Albuquerque - Revelações        2013              NaN   
1030875  Chico Albuquerque - Revelações        2013              NaN   

                     genres            directors  \
0        Action,Crime,Drama            nm0712540   
1        Action,Crime,Drama            nm0712540   
2        Action,Crime,Drama            nm0712540   
3        Action,Crime,Drama            nm0712540   
4        Action,Crime,Drama            nm0712540   
...                     ...                  ...   
1030871         Documentary  nm9272490,nm8349149   
1030872         Documentary  nm9272490,nm8349149   
1030873         Documentary  nm9272490,nm8349149   
1030874         Documentary  nm9272490,nm8349149   
1030875         Documentary  nm9272490,nm8349149   

                                         writers  ordering      nconst  \
0        nm0023551,nm1194313,nm0347899,nm1391276      10.0   nm0006210   
1        nm0023551,nm1194313,nm0347899,nm1391276       1.0   nm0474801   
2        nm0023551,nm1194313,nm0347899,nm1391276       2.0   nm0904537   
3        nm0023551,nm1194313,nm0347899,nm1391276       3.0   nm0756379   
4        nm0023551,nm1194313,nm0347899,nm1391276       4.0   nm0474876   
...                                          ...       ...         ...   
1030871                      nm8349149,nm9272490       5.0   nm9272490   
1030872                      nm8349149,nm9272490       6.0   nm8349149   
1030873                      nm8349149,nm9272490       7.0   nm9272489   
1030874                      nm8349149,nm9272490       8.0  nm10538638   
1030875                      nm8349149,nm9272490       9.0  nm10538635   

                category                   job  \
0               composer                   NaN   
1                  actor                   NaN   
2                actress                   NaN   
3                  actor                   NaN   
4                  actor                   NaN   
...                  ...                   ...   
1030871         director  supervising director   
1030872         director           co-director   
1030873  cinematographer                   NaN   
1030874  cinematographer                   NaN   
1030875  cinematographer                   NaN   

                              characters  
0                                    NaN  
1        ["Kundan S. Prasad","Bajrangi"]  
2            ["Munni","Laila-E-Aasmaan"]  
3                  ["Ganeshi N. Prasad"]  
4                   ["Dwarka N. Prasad"]  
...                                  ...  
1030871                              NaN  
1030872                              Na

In [12]:
third_join = second_join.merge(title_ratings_df, how='left', left_on='tconst', right_on='tconst')
third_join

tconst                   primary_title  \
0        tt0063540                       Sunghursh   
1        tt0063540                       Sunghursh   
2        tt0063540                       Sunghursh   
3        tt0063540                       Sunghursh   
4        tt0063540                       Sunghursh   
...            ...                             ...   
1030871  tt9916754  Chico Albuquerque - Revelações   
1030872  tt9916754  Chico Albuquerque - Revelações   
1030873  tt9916754  Chico Albuquerque - Revelações   
1030874  tt9916754  Chico Albuquerque - Revelações   
1030875  tt9916754  Chico Albuquerque - Revelações   

                         original_title  start_year  runtime_minutes  \
0                             Sunghursh        2013            175.0   
1                             Sunghursh        2013            175.0   
2                             Sunghursh        2013            175.0   
3                             Sunghursh        2013            175.0   
4                             Sunghursh        2013            175.0   
...                                 ...         ...              ...   
1030871  Chico Albuquerque - Revelações        2013              NaN   
1030872  Chico Albuquerque - Revelações        2013              NaN   
1030873  Chico Albuquerque - Revelações        2013              NaN   
1030874  Chico Albuquerque - Revelações        2013              NaN   
1030875  Chico Albuquerque - Revelações        2013              NaN   

                     genres            directors  \
0        Action,Crime,Drama            nm0712540   
1        Action,Crime,Drama            nm0712540   
2        Action,Crime,Drama            nm0712540   
3        Action,Crime,Drama            nm0712540   
4        Action,Crime,Drama            nm0712540   
...                     ...                  ...   
1030871         Documentary  nm9272490,nm8349149   
1030872         Documentary  nm9272490,nm8349149   
1030873         Documentary  nm9272490,nm8349149   
1030874         Documentary  nm9272490,nm8349149   
1030875         Documentary  nm9272490,nm8349149   

                                         writers  ordering      nconst  \
0        nm0023551,nm1194313,nm0347899,nm1391276      10.0   nm0006210   
1        nm0023551,nm1194313,nm0347899,nm1391276       1.0   nm0474801   
2        nm0023551,nm1194313,nm0347899,nm1391276       2.0   nm0904537   
3        nm0023551,nm1194313,nm0347899,nm1391276       3.0   nm0756379   
4        nm0023551,nm1194313,nm0347899,nm1391276       4.0   nm0474876   
...                                          ...       ...         ...   
1030871                      nm8349149,nm9272490       5.0   nm9272490   
1030872                      nm8349149,nm9272490       6.0   nm8349149   
1030873                      nm8349149,nm9272490       7.0   nm9272489   
1030874                      nm8349149,nm9272490       8.0  nm10538638   
1030875                      nm8349149,nm9272490       9.0  nm10538635   

                category                   job  \
0               composer                   NaN   
1                  actor                   NaN   
2                actress                   NaN   
3                  actor                   NaN   
4                  actor                   NaN   
...                  ...                   ...   
1030871         director  supervising director   
1030872         director           co-director   
1030873  cinematographer                   NaN   
1030874  cinematographer                   NaN   
1030875  cinematographer                   NaN   

                              characters  averagerating  numvotes  
0                                    NaN            7.0      77.0  
1        ["Kundan S. Prasad","Bajrangi"]            7.0      77.0  
2            ["Munni","Laila-E-Aasmaan"]            7.0      77.0  
3                  ["Ganeshi N. Prasad"]            7.0      77.0  
4                   ["Dwarka N. Prasad"]   

In [13]:
fourth_join = third_join.merge(title_akas_df, how='left', left_on='tconst', right_on='title_id')
fourth_join

tconst                   primary_title  \
0        tt0063540                       Sunghursh   
1        tt0063540                       Sunghursh   
2        tt0063540                       Sunghursh   
3        tt0063540                       Sunghursh   
4        tt0063540                       Sunghursh   
...            ...                             ...   
2975780  tt9916754  Chico Albuquerque - Revelações   
2975781  tt9916754  Chico Albuquerque - Revelações   
2975782  tt9916754  Chico Albuquerque - Revelações   
2975783  tt9916754  Chico Albuquerque - Revelações   
2975784  tt9916754  Chico Albuquerque - Revelações   

                         original_title  start_year  runtime_minutes  \
0                             Sunghursh        2013            175.0   
1                             Sunghursh        2013            175.0   
2                             Sunghursh        2013            175.0   
3                             Sunghursh        2013            175.0   
4                             Sunghursh        2013            175.0   
...                                 ...         ...              ...   
2975780  Chico Albuquerque - Revelações        2013              NaN   
2975781  Chico Albuquerque - Revelações        2013              NaN   
2975782  Chico Albuquerque - Revelações        2013              NaN   
2975783  Chico Albuquerque - Revelações        2013              NaN   
2975784  Chico Albuquerque - Revelações        2013              NaN   

                     genres            directors  \
0        Action,Crime,Drama            nm0712540   
1        Action,Crime,Drama            nm0712540   
2        Action,Crime,Drama            nm0712540   
3        Action,Crime,Drama            nm0712540   
4        Action,Crime,Drama            nm0712540   
...                     ...                  ...   
2975780         Documentary  nm9272490,nm8349149   
2975781         Documentary  nm9272490,nm8349149   
2975782         Documentary  nm9272490,nm8349149   
2975783         Documentary  nm9272490,nm8349149   
2975784         Documentary  nm9272490,nm8349149   

                                         writers  ordering_x      nconst  ...  \
0        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
1        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
2        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
3        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
4        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
...                                          ...         ...         ...  ...   
2975780                      nm8349149,nm9272490         5.0   nm9272490  ...   
2975781                      nm8349149,nm9272490         6.0   nm8349149  ...   
2975782                      nm8349149,nm9272490         7.0   nm9272489  ...   
2975783                      nm8349149,nm9272490         8.0  nm10538638  ...   
2975784                      nm8349149,nm9272490         9.0  nm10538635  ...   

        averagerating numvotes   title_id  ordering_y      title region  \
0                 7.0     77.0  tt0063540         1.0  Sangharsh     IN   
1                 7.0     77.0  tt0063540         2.0  Sunghursh    NaN   
2                 7.0     77.0  tt0063540         3.0  Sunghursh     IN   
3                 7.0     77.0  tt0063540         4.0  Sunghursh     IN   
4                 7.0     77.0  tt0063540         5.0  Sungharsh     IN   
...               ...      ...        ...         ...        ...    ...   
2975780           NaN      NaN        NaN         NaN        NaN    NaN   
2975781           NaN      NaN        NaN         NaN        NaN    NaN   
2975782           NaN      NaN        NaN         NaN        NaN    NaN   
2975783           NaN      NaN        NaN         NaN        NaN    NaN   
2975784           NaN      NaN        NaN         NaN        NaN    NaN   


In [14]:
fourth_join['title_id'].isna().sum()

134509

In [15]:
fourth_join.groupby('original_title').count()

tconst  primary_title  start_year  \
original_title                                                          
#1 Serial Killer                        30             30          30   
#5                                       9              9           9   
#50Fathers                               9              9           9   
#66                                     10             10          10   
#82 Summits                              5              5           5   
...                                    ...            ...         ...   
ärtico                                  50             50          50   
è solo questione di punti di vista       8              8           8   
è solo questione di tempo               15             15          15   
élèctions piege à cons                   6              6           6   
împetu                                   4              4           4   

                                    runtime_minutes  genres  directors  \
original_title                                                           
#1 Serial Killer                                 30      30         30   
#5                                                2       9          9   
#50Fathers                                        9       9          9   
#66                                              10      10         10   
#82 Summits                                       5       5          5   
...                                             ...     ...        ...   
ärtico                                           50      50         50   
è solo questione di punti di vista                8       8          8   
è solo questione di tempo                        15      15         15   
élèctions piege à cons                            0       6          6   
împetu                                            0       4          4   

                                    writers  ordering_x  nconst  category  \
original_title                                                              
#1 Serial Killer                         30          30      30        30   
#5                                        9           9       9         9   
#50Fathers                                9           9       9         9   
#66                                      10          10      10        10   
#82 Summits                               0           5       5         5   
...                                     ...         ...     ...       ...   
ärtico                                   50          50      50        50   
è solo questione di punti di vista        8           8       8         8   
è solo questione di tempo                15          15      15        15   
élèctions piege à cons                    0           6       6         6   
împetu                                    4           4       4         4   

                                    ...  averagerating  numvotes  title_id  \
original_title                      ...                                      
#1 Serial Killer                    ...             30        30        30   
#5                                  ...              2         2         2   
#50Fathers                          ...              0         0         9   
#66                                 ...             10        10        10   
#82 Summits                         ...              0         0         0   
...                                 ...            ...       ...       ...   
ärtico                              ...             50        50        50   
è solo questione di punti di vista  ...              0         0         8   
è solo questione di tempo           ...              0         0        15   
élèctions piege à cons              ...              0         0         0   
împetu                              ...              0         0         4   

                                    ordering_y  title  region  language  \
original_title                 

In [16]:
people_info_df

nconst         primary_name  birth_year  death_year  \
0       nm0061671    Mary Ellen Bauder         NaN         NaN   
1       nm0061865         Joseph Bauer         NaN         NaN   
2       nm0062070           Bruce Baum         NaN         NaN   
3       nm0062195         Axel Baumann         NaN         NaN   
4       nm0062798          Pete Baxter         NaN         NaN   
...           ...                  ...         ...         ...   
606643  nm9990381         Susan Grobes         NaN         NaN   
606644  nm9990690          Joo Yeon So         NaN         NaN   
606645  nm9991320       Madeline Smith         NaN         NaN   
606646  nm9991786  Michelle Modigliani         NaN         NaN   
606647  nm9993380       Pegasus Envoyé         NaN         NaN   

                                      primary_profession  \
0              miscellaneous,production_manager,producer   
1             composer,music_department,sound_department   
2                             miscellaneous,actor,writer   
3       camera_department,cinematographer,art_department   
4       production_designer,art_department,set_decorator   
...                                                  ...   
606643                                           actress   
606644                                           actress   
606645                                           actress   
606646                                          producer   
606647                             director,actor,writer   

                               known_for_titles  
0       tt0837562,tt2398241,tt0844471,tt0118553  
1       tt0896534,tt6791238,tt0287072,tt1682940  
2       tt1470654,tt0363631,tt0104030,tt0102898  
3       tt0114371,tt2004304,tt1618448,tt1224387  
4       tt0452644,tt0452692,tt3458030,tt2178256  
...                                         ...  
606643                                      NaN  
606644                      tt9090932,tt8737130  
606645                      tt8734436,tt9615610  
606646                                      NaN  
606647                                tt8743182  

[606648 rows x 6 columns]

In [17]:
fifth_join = fourth_join.merge(people_info_df, how='left', left_on='nconst', right_on='nconst')
fifth_join

tconst                   primary_title  \
0        tt0063540                       Sunghursh   
1        tt0063540                       Sunghursh   
2        tt0063540                       Sunghursh   
3        tt0063540                       Sunghursh   
4        tt0063540                       Sunghursh   
...            ...                             ...   
2975780  tt9916754  Chico Albuquerque - Revelações   
2975781  tt9916754  Chico Albuquerque - Revelações   
2975782  tt9916754  Chico Albuquerque - Revelações   
2975783  tt9916754  Chico Albuquerque - Revelações   
2975784  tt9916754  Chico Albuquerque - Revelações   

                         original_title  start_year  runtime_minutes  \
0                             Sunghursh        2013            175.0   
1                             Sunghursh        2013            175.0   
2                             Sunghursh        2013            175.0   
3                             Sunghursh        2013            175.0   
4                             Sunghursh        2013            175.0   
...                                 ...         ...              ...   
2975780  Chico Albuquerque - Revelações        2013              NaN   
2975781  Chico Albuquerque - Revelações        2013              NaN   
2975782  Chico Albuquerque - Revelações        2013              NaN   
2975783  Chico Albuquerque - Revelações        2013              NaN   
2975784  Chico Albuquerque - Revelações        2013              NaN   

                     genres            directors  \
0        Action,Crime,Drama            nm0712540   
1        Action,Crime,Drama            nm0712540   
2        Action,Crime,Drama            nm0712540   
3        Action,Crime,Drama            nm0712540   
4        Action,Crime,Drama            nm0712540   
...                     ...                  ...   
2975780         Documentary  nm9272490,nm8349149   
2975781         Documentary  nm9272490,nm8349149   
2975782         Documentary  nm9272490,nm8349149   
2975783         Documentary  nm9272490,nm8349149   
2975784         Documentary  nm9272490,nm8349149   

                                         writers  ordering_x      nconst  ...  \
0        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
1        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
2        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
3        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
4        nm0023551,nm1194313,nm0347899,nm1391276        10.0   nm0006210  ...   
...                                          ...         ...         ...  ...   
2975780                      nm8349149,nm9272490         5.0   nm9272490  ...   
2975781                      nm8349149,nm9272490         6.0   nm8349149  ...   
2975782                      nm8349149,nm9272490         7.0   nm9272489  ...   
2975783                      nm8349149,nm9272490         8.0  nm10538638  ...   
2975784                      nm8349149,nm9272490         9.0  nm10538635  ...   

        region language     types                   attributes  \
0           IN       hi       NaN  alternative transliteration   
1          NaN      NaN  original                          NaN   
2           IN      NaN       NaN                          NaN   
3           IN       hi       NaN  alternative transliteration   
4           IN       hi       NaN         alternative spelling   
...        ...      ...       ...                          ...   
2975780    NaN      NaN       NaN                          NaN   
2975781    NaN      NaN       NaN                          NaN   
2975782    NaN      NaN       NaN                          NaN   
2975783    NaN      NaN       NaN                          NaN   
2975784    NaN      NaN       NaN                          NaN   

         is_original_title            primary_name  birth_year death_year  \
0                      0.0    

In [18]:
fifth_join.groupby('original_title')

In [19]:
fifth_join.isna().sum()

tconst                      0
primary_title               0
original_title             32
start_year                  0
runtime_minutes        280635
genres                  32591
directors               20435
writers                265826
ordering_x               2812
nconst                   2812
category                 2812
job                   2231222
characters            1824758
averagerating          519637
numvotes               519637
title_id               134509
ordering_y             134509
title                  134509
region                 595279
language              2596060
types                 1423140
attributes            2839834
is_original_title      134534
primary_name             3302
birth_year            1833225
death_year            2928068
primary_profession      95934
known_for_titles        49511
dtype: int64

### Join the large 'tconst' dataframe with the movie_gross_df. 

Make the join three times, once on each of the columns containing the movie title.

There's probably a better way to do this, but that's just how the cookie crumbles.

In [25]:
sixth_join = fifth_join.merge(movie_gross_df, how='left', left_on='primary_title', right_on='title')

In [26]:
sixth_join.shape

(2975862, 33)

In [27]:
sixth_join = fifth_join.merge(movie_gross_df, how='left', left_on='original_title', right_on='title')

In [28]:
sixth_join.shape

(2975785, 33)

In [29]:
sixth_join = fifth_join.merge(movie_gross_df, how='left', left_on='title', right_on='title')

In [30]:
sixth_join.shape

(2975817, 32)

In [60]:
list(sixth_join.columns)

['tconst',
 'primary_title',
 'original_title',
 'start_year',
 'runtime_minutes',
 'genres',
 'directors',
 'writers',
 'ordering_x',
 'nconst',
 'category',
 'job',
 'characters',
 'averagerating',
 'numvotes',
 'title_id',
 'ordering_y',
 'title',
 'region',
 'language',
 'types',
 'attributes',
 'is_original_title',
 'primary_name',
 'birth_year',
 'death_year',
 'primary_profession',
 'known_for_titles',
 'studio',
 'domestic_gross',
 'foreign_gross',
 'year']

In [36]:
sixth_join.groupby('title').mean().head(50)

start_year  \
title                                                            
!Women Art Revolution                                   2010.0   
# Pire soirée                                           2017.0   
#1 Serial Killer                                        2013.0   
#211                                                    2018.0   
#5                                                      2015.5   
#50Fathers                                              2015.0   
#66                                                     2015.0   
#9                                                      2019.0   
#ActorsWanted                                           2011.0   
#Afskåret                                               2016.0   
#BKKY                                                   2016.0   
#BeRobin the Movie                                      2015.0   
#BeastMode                                              2018.0   
#Beings                                                 2015.0   
#Bicyclediary                                           2018.0   
#Bitch, les filles et la violence                       2015.0   
#BreakfastInBedTour                                     2013.0   
#BuscandoaInés                                          2016.0   
#Cannibal Island                                        2016.0   
#Captured                                               2017.0   
#Chef                                                   2014.0   
#CoEdConfessions                                        2017.0   
#DigitalLivesMatter                                     2016.0   
#Disneyland60                                           2015.0   
#Enough                                                 2015.0   
#EraDosGigantes                                         2016.0   
#EradosGigantes                                         2017.0   
#Escape                                                 2019.0   
#Ewankosau saranghaeyo                                  2015.0   
#Exorcismo                                              2015.0   
#Fafatl                                                 2017.0   
#Female Pleasure                                        2018.0   
#Flora63                                                2017.0   
#FollowFriday                                           2016.0   
#Followme                                               2019.0   
#FromJennifer                                           2017.0   
#Gadhvi                                                 2018.0   
#Gadhvi (He thought he was Gandhi)                      2018.0   
#Generation Bieber                                      2018.0   
#Hashtag: A Millennial Guide to Why Everything ...      2017.0   
#HearMe                                                 2016.0   
#Hommes: True Story of the Thieves' World               2015.0   
#Horror                                                 2015.0   
#Iamhere                                                2020.0   
#Indignados                                             2011.0   
#IrishGameDev 2014                                      2015.0   
#Jézus - Apám nevében                                   2017.0   
#Like                                                   2019.0   
#Lucky Number                                           2015.0   
#MeToo                                                  2019.0   

                                                    runtime_minutes  \
title                                                                 
!Women Art Revolution                                     83.000000   
# Pire soirée                                            101.000000   
#1 Serial Killer                                          87.000000   
#211                                                      86.000000   
#5                                                       114.666667   
#50Fathers                                               132.000000   
#66                                                

### Export to CSV

In [37]:
sixth_join.to_csv('bigol_df.csv')

### Remove all entries without a domestic gross & foreign gross (in progress)

In [59]:
sixth_join.dropna(subset=['domestic_gross', 'foreign_gross'], how='any').groupby('primary_title').mean()


start_year  runtime_minutes  ordering_x  \
primary_title                                                         
'71                             2014.0             99.0         5.5   
1                               2019.0            100.0         4.0   
10 Cloverfield Lane             2016.0            103.0         5.5   
102 Not Out                     2018.0            102.0         5.5   
11-11-11                        2011.0             90.0         5.5   
...                                ...              ...         ...   
Zero Dark Thirty                2012.0            157.0         5.5   
Zookeeper                       2011.0            102.0         5.5   
Zoolander 2                     2016.0            101.0         5.5   
Zootopia                        2016.0            108.0         5.5   
xXx: Return of Xander Cage      2017.0            107.0         5.5   

                            averagerating  numvotes  ordering_y  \
primary_title                                                     
'71                                   7.2   46103.0   12.111111   
1                                     9.2      13.0    2.000000   
10 Cloverfield Lane                   7.2  260383.0   17.800000   
102 Not Out                           7.5    4802.0    2.000000   
11-11-11                              4.0   11712.0    7.333333   
...                                   ...       ...         ...   
Zero Dark Thirty                      7.4  251072.0   16.250000   
Zookeeper                             5.2   52396.0   12.666667   
Zoolander 2                           4.7   59914.0   12.000000   
Zootopia                              8.0  383446.0   34.600000   
xXx: Return of Xander Cage            5.2   77976.0    7.000000   

                            is_original_title   birth_year  death_year  \
primary_title                                                            
'71                                  0.111111  1977.666667         NaN   
1                                    0.000000  1951.666667         NaN   
10 Cloverfield Lane                  0.200000  1972.833333         NaN   
102 Not Out                          0.500000  1947.000000         NaN   
11-11-11                             0.166667  1966.400000         NaN   
...                                       ...          ...         ...   
Zero Dark Thirty                     0.125000  1969.800000         NaN   
Zookeeper                            0.333333  1968.571429         NaN   
Zoolander 2                          0.090909  1966.700000      2004.0   
Zootopia                             0.200000  1972.125000         NaN   
xXx: Return of Xander Cage           0.000000  1969.285714         NaN   

                            domestic_gross    year  
primary_title                                       
'71                              1300000.0  2015.0  
1                               27800000.0  2010.0  
10 Cloverfield Lane             72100000.0  2016.0  
102 Not Out                      1300000.0  2018.0  
11-11-11                           32800.0  2011.0  
...                                    ...     ...  
Zero Dark Thirty                95700000.0  2012.0  
Zookeeper                       80400000.0  2011.0  
Zoolander 2                     28800000.0  2016.0  
Zootopia                       341300000.0  2016.0  
xXx: Return of Xander Cage      44900000.0  2017.0  

[1760 rows x 11 columns]

### Questions that I think may be worth exploring:

1. Should we try to fill in missing box office numbers? If yes, we could see if domestic and foreign grosses usually correlate. If they do, we could come up with an average y=ax relationship and fill in missing values that way. If not, there are 1760 good records we have that I can make a clean dataframe from and we can just use that for gross returns.

2. What are the top 100 films by gross returns over the time period? 
    a. What are their genres? If we aggregate, what are the three most common genres?
    b. Who directed them?
    c. Who stars?
    d. Which studios made them?
    e. What are their runtimes? If we aggregate, what is the mean? How dispersed?
    
3. What are the bottom 100 films by gross returns? Same subquestions.

4. Many movie studios have their own "mini-studios" to compete with the independent and indy-filmmaking scene. Are the relative profits of big budget movies higher or lower than low budget movies? Are the relative profits of one category more dispersed than the other category? Depending on the answers, we might recommend to Microsoft to either, indeed, create a "mini-studio" of their own, or to forego doing so.

5. Do studios who make a high amount of movies per year make more profit (relative to budgets) than studios who make a small amount of movies per year?

6. Do critically acclaimed movies correlate to higher profits? If no, I think it's worth pointing this out, and specifally warning them against chasing after critical acclaim. 